extension/js/common/core/crypto/pgp/openpgp-key.ts

```diff
@@ -618,127 +610,150 @@ export class OpenPGPKey {
+        opgp.enums.read(opgp.enums.publicKey, opgp.enums.publicKey.eddsa),
+      ].includes(key.getAlgorithmInfo().algorithm)
+    ) {
       return 0; // disallow encryption for these algorithms
     }
-    return signature.keyFlags[0] & (opgp.enums.keyFlags.encrypt_communication | opgp.enums.keyFlags.encrypt_storage);
+    return verifiedSignature.keyFlags[0] & (opgp.enums.keyFlags.encryptCommunication | opgp.enums.keyFlags.encryptStorage);
   };
 
-  private static getValidSigningKeyPacketFlags = (
-    keyPacket: OpenPGP.packet.PublicKey | OpenPGP.packet.SecretKey,
-    signature: OpenPGP.packet.Signature
-  ): OpenPGP.enums.keyFlags => {
-    if (!signature.keyFlags || !signature.verified || signature.revoked !== false) {
+  private static getValidSigningKeyPacketFlags = (key: OpenPGP.Key | OpenPGP.Subkey, verifiedSignature: OpenPGP.SignaturePacket): OpenPGP.enums.keyFlags => {
+    if (!verifiedSignature.keyFlags || verifiedSignature.revoked !== false) {
       // Sanity check
       return 0;
     }
-    if ([opgp.enums.publicKey.rsa_encrypt, opgp.enums.publicKey.elgamal, opgp.enums.publicKey.ecdh].includes(keyPacket.algorithm)) {
+    if (
+      [
+        opgp.enums.read(opgp.enums.publicKey, opgp.enums.publicKey.rsaEncrypt),
+        opgp.enums.read(opgp.enums.publicKey, opgp.enums.publicKey.elgamal),
+        opgp.enums.read(opgp.enums.publicKey, opgp.enums.publicKey.ecdh),
+      ].includes(key.getAlgorithmInfo().algorithm)
+    ) {
       return 0; // disallow signing for these algorithms
     }
-    return signature.keyFlags[0] & (opgp.enums.keyFlags.sign_data | opgp.enums.keyFlags.certify_keys);
+    return verifiedSignature.keyFlags[0] & (opgp.enums.keyFlags.signData | opgp.enums.keyFlags.certifyKeys);
   };
 
-  private static getSubKeySigningFlags = async (key: OpenPGP.key.Key, subKey: OpenPGP.key.SubKey): Promise<OpenPGP.enums.keyFlags> => {
+  private static getSubKeySigningFlags = async (key: OpenPGP.PrivateKey | OpenPGP.PublicKey, subKey: OpenPGP.Subkey): Promise<OpenPGP.enums.keyFlags> => {
     const primaryKey = key.keyPacket;
     // await subKey.verify(primaryKey);
     const dataToVerify = { key: primaryKey, bind: subKey.keyPacket };
-    const date = new Date();
-    const bindingSignature = await OpenPGPKey.getLatestValidSignature(
-      subKey.bindingSignatures,
-      primaryKey,
-      opgp.enums.signature.subkey_binding,
-      dataToVerify,
-      date
-    );
+    const bindingSignature = await OpenPGPKey.getLatestValidSignature(subKey.bindingSignatures, primaryKey, opgp.enums.signature.subkeyBinding, dataToVerify);
     if (
       bindingSignature &&
       bindingSignature.embeddedSignature &&
-      (await OpenPGPKey.getLatestValidSignature([bindingSignature.embeddedSignature], subKey.keyPacket, opgp.enums.signature.key_binding, dataToVerify, date))
+      (await OpenPGPKey.getLatestValidSignature([bindingSignature.embeddedSignature], subKey.keyPacket, opgp.enums.signature.keyBinding, dataToVerify))
```

Comment:
I'm not sure and I'm ok to not worry about it if it passes existing tests. However, it would be good to file it for another issue to consider further.